Create rule S6542: Any should not be used as a type hint (#1665)
SonarSource/rspec: rules/S6542/python/rule.adoc

== Why is this an issue?

The type hint `Any` represents any possible Python type, on which all operations are possible. This conveys the information that all operations could be possible on a value annotated with `Any`, making the value dynamically-typed.

> `Any` as a type hint does not provide any information about the expected type of the variable or parameter, essentially losing the benefits of type hinting. This can make the code less clear and harder to understand and will reduce the code insight capabilities of IDEs and static analysis tools.

Reviewer: a suggestion to avoid repeating "any"
```suggestion
`Any` as a type hint provides no information about the expected type of the variable or parameter, essentially losing the benefits of type hinting. This can make the code less clear and harder to understand and will reduce the code insight capabilities of IDEs and static analysis tools.
```
